tools/cell_census_builder/validate.py

```diff
@@ -66,101 +66,115 @@ def validate_all_soma_objects_exist(soma_path: str, experiment_builders: List[Ex
         |   +-- homo_sapiens: soma.Experiment
         |   +-- mus_musculus: soma.Experiment
     """
-    census = soma.Collection(soma_path, context=SOMA_TileDB_Context())
-    assert census.exists() and census.soma_type == "SOMACollection"
-    assert "cxg_schema_version" in census.metadata and census.metadata["cxg_schema_version"] == CXG_SCHEMA_VERSION
-    assert (
-        "census_schema_version" in census.metadata and census.metadata["census_schema_version"] == CENSUS_SCHEMA_VERSION
-    )
-    assert "created_on" in census.metadata and datetime.fromisoformat(census.metadata["created_on"])
-    assert "git_commit_sha" in census.metadata
-
-    for name in [CENSUS_INFO_NAME, CENSUS_DATA_NAME]:
-        assert name in census
-        assert census[name].soma_type == "SOMACollection"
-        assert census[name].exists()
-
-    census_info = census[CENSUS_INFO_NAME]
-    for name in [CENSUS_DATASETS_NAME, CENSUS_SUMMARY_NAME, CENSUS_SUMMARY_CELL_COUNTS_NAME]:
-        assert name in census_info, f"`{name}` missing from census_info"
-        assert census_info[name].soma_type == "SOMADataFrame"
-        assert census_info[name].exists()
-
-    assert sorted(census_info[CENSUS_DATASETS_NAME].keys()) == sorted(CENSUS_DATASETS_COLUMNS + ["soma_joinid"])
-    assert sorted(census_info[CENSUS_SUMMARY_CELL_COUNTS_NAME].keys()) == sorted(
-        list(CENSUS_SUMMARY_CELL_COUNTS_COLUMNS) + ["soma_joinid"]
-    )
-    assert sorted(census_info[CENSUS_SUMMARY_NAME].keys()) == sorted(["label", "value", "soma_joinid"])
-
-    # there should be an experiment for each builder
-    census_data = census[CENSUS_DATA_NAME]
-    for eb in experiment_builders:
+    with soma.Collection.open(soma_path, context=SOMA_TileDB_Context()) as census:
```

Comment:
you could also use `soma.open()` given that you explicitly check the type in the next line